ray-curator/ray_curator/stages/text/deduplication/removal_workflow.py

```diff
@@ -0,0 +1,163 @@
+                )
+            )
+        else:
+            fields_to_ignore = ["input_path", "input_files_per_partition", "input_blocksize", "input_file_extensions"]
+            logger.warning(f"Initial tasks provided, ignoring {fields_to_ignore}")
+
+        if self.input_filetype == "parquet":
+            from ray_curator.stages.text.io.reader.parquet import ParquetReaderStage
+
+            read_stage = ParquetReaderStage
+        elif self.input_filetype == "jsonl":
+            from ray_curator.stages.text.io.reader.jsonl import JsonlReaderStage
+
+            read_stage = JsonlReaderStage
+        else:
+            msg = f"Invalid input filetype: {self.input_filetype}"
+            raise ValueError(msg)
+
+        stages.append(
+            read_stage(
+                fields=self.input_fields,
+                read_kwargs=self.input_kwargs,
+                _generate_ids=False,
+                _assign_ids=self.id_generator_path is not None,
+            )
+        )
+
+        stages.append(
+            TextDuplicatesRemovalStage(
+                ids_to_remove_path=self.ids_to_remove_path,
+                id_field=self.input_id_field,
+                duplicate_id_field=self.ids_to_remove_duplicate_id_field,
+                read_kwargs=self.ids_to_remove_read_kwargs,
+            )
+        )
+
+        if self.output_filetype == "parquet":
+            from ray_curator.stages.text.io.writer.parquet import ParquetWriter
+
+            write_stage = ParquetWriter
+        elif self.output_filetype == "jsonl":
+            from ray_curator.stages.text.io.writer.jsonl import JsonlWriter
+
+            write_stage = JsonlWriter
+        else:
+            msg = f"Invalid output filetype: {self.output_filetype}"
+            raise ValueError(msg)
+
+        stages.append(
+            write_stage(
+                path=self.output_path,
+                **({"file_extension": self.output_file_extension} if self.output_file_extension else {}),
+                write_kwargs=self.output_kwargs,
+                fields=self.output_fields,
+                **({"mode": self.output_mode} if self.output_mode else {}),
+            )
+        )
+
+        return stages
+
+    def run(
+        self, executor: Optional["BaseExecutor"] = None, initial_tasks: list[FileGroupTask] | None = None
+    ) -> list[FileGroupTask] | None:
+        stages = self._generate_stages(initial_tasks)
+
+        if executor is None:
+            from ray_curator.backends.xenna import XennaExecutor
+
+            executor = XennaExecutor()
+
+        if self.id_generator_path is not None:
+            from ray_curator.stages.deduplication.id_generator import (
+                create_id_generator_actor,
+                kill_id_generator_actor,
+            )
+
+            create_id_generator_actor(self.id_generator_path, storage_options=self.id_generator_storage_options)
+            output = executor.execute(stages, initial_tasks=initial_tasks)
```

Comment:
Nit: Prefer creating a pipeline and calling `pipeline.run(executor, initial_tasks)`